core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -467,6 +467,33 @@ public void clearInvalidSegments(CarbonTable carbonTable, List<Segment> segments
 
   }
 
+  public void refreshSegmentCacheIfRequired(CarbonTable carbonTable,
+      SegmentUpdateStatusManager updateStatusManager, List<Segment> filteredSegmentToAccess)
+      throws IOException {
+    List<String> toBeCleanedSegments = new ArrayList<>();
+    for (Segment filteredSegment : filteredSegmentToAccess) {
```

Comment:
use `map` instead of `for`

Reply:
u mean filteredSegmentToAccess.map {...}???